### PR TITLE
Create About Kuzya *.ui, *.cpp and *.hpp

### PR DESCRIPTION
**Pull request by dmykuliak-cat**
Created UI for About Kuzya, and code which contained in slotAbout() moved to *.hpp and *.cpp

### CODE REVIEW

File: src/kuzya.cpp [hunk 1]
```diff
@@ -52,7 +47,7 @@
 #include "helpbrowser.h"
 #include "translator.h"
 #include "version.h"
-
+#include "kuzyaabout.hpp"
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Why it has extension .hpp?

Comment (alexchmykhalo) [reply src/kuzya.cpp]: I'll also suggest to find better name for class and files (aboutdialog.*, aboutkuzya.*) -it sounds a bit more natural.

File: src/kuzya.cpp [hunk 2]
```diff
@@ -1034,44 +1029,11 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
-    scrollAreaAuthors->setWidget(textInfLabel);
-    layout->addWidget(scrollAreaAuthors);
-    information->setWindowIcon(QIcon(QDir::currentPath()+":/../../src/images/kuzya.png"));
-
-    information->exec();
-
-    delete information;
+    KuzyaAbout kuzya;
+    kuzya.checkFile();
+    kuzya.readAuthors();
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: It is ok to have all this methods to separate diffetent operations that dialog performs. But I see no point to make them public and call them here. What if someone will not know in that order to call all that methods? What if someone forgot to call one of them?

I sugest make them private and call in constructor.

Comment (alexchmykhalo) [reply src/kuzya.cpp]: Another thing that I see - that KuzyaAbout dialog has a number of dependencies. Kuzya version and build date are used from "version.h", also path to AUTHORS file is hardcoded. 

I suggest to make setter methods that will take version, build date and path to file with authors information. Then KuzyaAbout dialog will be more independent class that will be responsible only for displaying information about Kuzua IDE and will not know all application related details (such as AUTHORS file location, how version and build date are presented and where stored)

Comment (alexchmykhalo) [reply src/kuzya.cpp]: It is event better to pass Kuzya version,  build date and  path to AUTHORS file - as a constructor parameters. KuzyaAbout dialog will not show correct information if we not set all that, right?

File: src/kuzyaabout.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  explicit KuzyaAbout(QWidget *parent = 0);
+  void checkFile();
+  void readAuthors();
+  void setVerAndDate();
+  void fillWindow();
+  ~KuzyaAbout();
+
+private:
+  Ui::KuzyaAbout *ui;
+  const QString *fileName = new QString(":/AUTHORS.txt");
```
Comment (alexchmykhalo) [inline src/kuzyaabout.hpp]: Please, use m prefix for member names.

File: src/kuzyaabout.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+//set Kuzia version and build date
+void KuzyaAbout::setVerAndDate()
+{
+  QVersionNumber verKuzia(MAJORVER, MINORVER, REVISION);
+  QDate buildDate(BUILD_YEAR, BUILD_MONTH, BUILD_DAY);
+  ui->textInfLabel->setText(textInf->arg(verKuzia.toString(), buildDate.toString("MMMM d yyyy")));
+}
+
+//fill Dialog window
+void KuzyaAbout::fillWindow()
```
Comment (alexchmykhalo) [inline src/kuzyaabout.cpp]: I think you can make all this in Designer.

File: src/kuzya.cpp (hunk 2 above)
Comment (dmykuliak-cat) [reply src/kuzya.cpp]: you're right!

File: src/aboutkuzya.cpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  ui->layout = new QVBoxLayout(this);
+  this->setLayout(ui->layout);
+  ui->textInfLabel = new QLabel(this);
+  ui->imgLabel = new QLabel(this);
+
+  checkFile();
+  readAuthors();
+  setVerAndDate();
+
+  exec();
```
Comment (alexchmykhalo) [inline src/aboutkuzya.cpp]: Never do exec() or other such things from the constructor. Developer should have an opportunity to open dialog when it is needed. That is why exec() is a separate method.

Please, move that call to place where instance is created.

File: src/aboutkuzya.cpp [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+//checks file
+void AboutKuzya::checkFile()
+{
+  mFile = new QFile (*mFileName);
+
+  if(!QFile::exists(*mFileName))
+  {
+      qCritical()<< "File doesn't exist"<<*mFileName<<endl;
```
Comment (alexchmykhalo) [inline src/aboutkuzya.cpp]: You could return from method if file does not exist - it will not open anyway.

File: src/aboutkuzya.cpp [hunk 7]
```diff
@@ -0,0 +1,61 @@
+#include "aboutkuzya.h"
+#include "ui_aboutkuzya.h"
+
+#include <QDebug>
+#include <QDir>
+
+AboutKuzya::AboutKuzya(QWidget *parent, QVersionNumber *verKuzia, QDate *buildDate, QString *fileName) :
```
Comment (alexchmykhalo) [inline src/aboutkuzya.cpp]: You need to place "QWidget *parent" argument in the end of argument list. Then you could make it default (=nullptr).

File: src/aboutkuzya.h [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+namespace Ui {
+  class AboutKuzya;
+}
+
+class AboutKuzya : public QDialog
+{
+  Q_OBJECT
+
+public:
+  explicit AboutKuzya(QWidget *parent = 0, QVersionNumber *verKuzia = 0, QDate *buildDate = 0, QString *fileName = 0);
```
Comment (alexchmykhalo) [inline src/aboutkuzya.h]: Why you need to pass all this arguments by pointer?

**Review by alexchmykhalo — COMMENTED** [review]
Please, see new review comments.

**Comment by dmykuliak-cat** [comment]
Check it out

File: src/aboutkuzya.cpp [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  if (!mFile->open(QIODevice::ReadOnly | QIODevice::Text))
+  {
+      return;
+    }
+}
+
+//reading AUTHORS file
+void AboutKuzya::readAuthors()
+{
+  mIn = new QTextStream(mFile);
```
Comment (alexchmykhalo) [inline src/aboutkuzya.cpp]: Read authors will be called regardless of checkFile() call state. I suggest to merge this two nethods to one. In that new method you could check for authors file and skip reading if it is not exist.

File: src/aboutkuzya.cpp [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  mTextInf = new QString;
```
Comment (alexchmykhalo) [inline src/aboutkuzya.cpp]: Why you need to create QString, QTextStream and QFile on a heap? Please, use stack as much as possible.

File: src/aboutkuzya.h [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  Q_OBJECT
+
+public:
+  AboutKuzya(QVersionNumber verKuzia, QDate buildDate, QString fileName, QWidget *parent = 0);
+  ~AboutKuzya();
+
+private:
+  Ui::AboutKuzya *ui;
+  QString *mFileName;
+  QFile *mFile;
```
Comment (alexchmykhalo) [inline src/aboutkuzya.h]: Again, you do not need all this pointers. You could create string, stream and file objects locally in your method.

**Review by alexchmykhalo — CHANGES_REQUESTED** [review]
Please, see review comments.

**Review by alexchmykhalo — APPROVED** [review]

**Pull request merged**